### DISCUSSION TITLE
Why?

### DISCUSSION THREAD

**Discussion by DeadSecTr** [fast4x/RiMusic#872] [Ideas]
RiMusic was a really good music app before 3-4 updates ago..because it really focused on listening to music and it was on "Point"

And not like now, at the moment, stuffed with unnecessary features and adjustments that nobody needs!

That's why it's become so unreliable lately and crashes regularly...

What's definitely missing... is a synchronization function with Piped!
It would make it much easier for users who want to switch to Rimusic or vice versa...

I've recently started using Harmony Music, which I really like because it supports Piped sync and only has the essential settings/functions that a music player/streamer needs!

I appreciate the effort put into this work a lot. I know it took a lot of work and time to code it... but please, developer, don't implement everything that is desired, as it deviates from what RiMusic was intended to be or was.

This is my personal opinion and experience report...

**Comment by fast4x** [comment]
Piped require a lot of time to develop it. Simple.

**Comment by ikanakova** [comment]
On the other hand, I like @fast4x's approach of trying to please everyone as much as possible. I've seen where it leads when a developer doesn't care about his users.
I'm a bit upset that RiMusic requires a lot more permissions than ViMusic, but I understand why that is and it's explained in the FAQ, so in the end I don't mind.
On the contrary, I find Piped a complete uselessness.
We all just have different preferences and you can never please everyone...
I'm very glad that @fast4x has its limits, that it doesn't plan to add all sorts of trackers eg from Google just to please users who want account logins.

What really annoys me is how some users here are pretty aggressively demanding various features.

**Reply by fast4x** [reply]
> On the other hand, I like @fast4x's approach of trying to please everyone as much as possible. I've seen where it leads when a developer doesn't care about his users.
> I'm a bit upset that RiMusic requires a lot more permissions than ViMusic, but I understand why that is and it's explained in the FAQ, so in the end I don't mind.
> On the contrary, I find Piped a complete uselessness.
> We all just have different preferences and you can never please everyone...
> I'm very glad that @fast4x has its limits, that it doesn't plan to add all sorts of trackers eg from Google just to please users who want account logins.
> 
> What really annoys me is how some users here are pretty aggressively demanding various features. 

Thanks, just one thing, RiMusic requires the same permissions declared in the ViMusic project manifest file, it only added those required by Android 14 and the microphone for the visualizer.

**Reply by ikanakova** [reply]
> > On the other hand, I like @fast4x's approach of trying to please everyone as much as possible. I've seen where it leads when a developer doesn't care about his users.
> > I'm a bit upset that RiMusic requires a lot more permissions than ViMusic, but I understand why that is and it's explained in the FAQ, so in the end I don't mind.
> > On the contrary, I find Piped a complete uselessness.
> > We all just have different preferences and you can never please everyone...
> > I'm very glad that @fast4x has its limits, that it doesn't plan to add all sorts of trackers eg from Google just to please users who want account logins.
> > What really annoys me is how some users here are pretty aggressively demanding various features.
> 
> Thanks, just one thing, RiMusic requires the same permissions declared in the ViMusic project manifest file, it only added those required by Android 14 and the microphone for the visualizer.

And local files playback.
I know what they're for and have no problem with it, I was just startled at first that according to [IzzyOnDroid](https://android.izzysoft.de/applists/category/named/multimedia_music_network), ViMusic has 6 permissions and RiMusic has 12.

**Comment by bartoostveen** [comment]
> Piped require a lot of time to develop it. Simple.

It took me an afternoon or two, it can't possibly take that long, right?

Edit: didn't mean to start a new thread but apparently 'quote reply' on GitHub mobile means reply in a new comment.

**Reply by fast4x** [reply]
> > Piped require a lot of time to develop it. Simple.
> 
> It took me an afternoon or two, it can't possibly take that long, right?

It means you’re more predisposed to that kind of development.

**Comment by fast4x** [comment]
I understand, for me two afternoons is a full working day, at the moment I have other priorities to work on.

**Reply by DeadSecTr** [reply]
"I completely understand, that should not be taken as an attack from me, it's just what I think and have felt from using the app!

But instead of adding new features, which could overload it in the end, I believe one should optimize the existing ones...

Regarding Piped Sync: 

With Piped, many users could easily switch to RiMusic and take their playlists with them, which in turn would bring more users>awareness... as they wouldn't have to search for and save each individual song in RiMusic playlists :)

I appreciate the work you've put into RiMusic... I just have concerns that it might end up being too complex and might abandon the app...

**Comment by ikanakova** [comment]
I admire you @fast4x that you don't mind when some users open the same duplicate issue over and over again. You put so much work into this app to make it great, and they can't be bothered to find if their feature hasn't been requested by someone before them.

**Reply by fast4x** [reply]
Thank you, i prefer not to report duplicates because it is a work that takes time and leaves a bit of a bitter taste in the mouth to those who add the request. In this way, people will express themselves more easily and when they see that their request has been considered as the other double, they will feel involved.

**Reply by ikanakova** [reply]
I think I get it. But it seems to me that they are opening a new issue because they think that if they start a new one, they will get higher on the priority list than if they post a comment on an existing one, which is nonsense and makes it harder to find for those who don't want to make duplicates.

**Reply by fast4x** [reply]
But the duplicate is also an opportunity to understand user preferences.

**Reply by OrangeZXZ** [reply]
> But the duplicate is also an opportunity to understand user preferences.

also to get ideas

**Reply by fast4x** [reply]
Sure

**Comment by siggi1984** [comment]
> RiMusic was a really good music app before 3-4 updates ago..because it really focused on listening to music and it was on "Point"
> 
> And not like now, at the moment, stuffed with unnecessary features and adjustments that nobody needs!
> 
> That's why it's become so unreliable lately and crashes regularly...
> 
> What's definitely missing... is a synchronization function with Piped! It would make it much easier for users who want to switch to Rimusic or vice versa...
> 
> I've recently started using Harmony Music, which I really like because it supports Piped sync and only has the essential settings/functions that a music player/streamer needs!
> 
> I appreciate the effort put into this work a lot. I know it took a lot of work and time to code it... but please, developer, don't implement everything that is desired, as it deviates from what RiMusic was intended to be or was.
> 
> This is my personal opinion and experience report...

If you are looking for a simpler app. Then you might want to take a look at this fork, which doesn't have Piped integration (but what isn't may come)

https://github.com/DanielSevillano/music-you
